### PR TITLE
Enforce code style rule IDE0055

### PR DESCRIPTION
**Pull request by RoosterDragon**
Enforce IDE0055 Formatting rule https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0055

This rule no longer appears to be buggy, so enforce it. Some of the automated fixes are adjusted in order to improve the result. `#pragma` directives have no option to control indentation, so remove them where possible.

The default options are generally workable for us. We have a mix of existing styles for `csharp_indent_case_contents_when_block`, but using the non-default results in a smaller diff.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeployWithCharge.cs [hunk 1]
```diff
@@ -217,8 +217,10 @@ void ITick.Tick(Actor self)
 						chargedToken = self.RevokeCondition(chargedToken);
 				}
 				else
+				{
 					if (chargedToken == Actor.InvalidConditionToken)
 						chargedToken = self.GrantCondition(Info.ChargedCondition);
+				}
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeployWithCharge.cs:223]: this was supposed to be allowed 🤔

Comment (RoosterDragon) [reply OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeployWithCharge.cs:223]: The rule wants:
```c#
if (chargeTick < (deployed ? deployedChargeThreshold : chargeThreshold))
{
	if (chargedToken != Actor.InvalidConditionToken)
		chargedToken = self.RevokeCondition(chargedToken);
}
else if (chargedToken == Actor.InvalidConditionToken)
	chargedToken = self.GrantCondition(Info.ChargedCondition);
```
but I felt keeping the visual symmetry was a better outcome:
```c#
if (chargeTick < (deployed ? deployedChargeThreshold : chargeThreshold))
{
	if (chargedToken != Actor.InvalidConditionToken)
		chargedToken = self.RevokeCondition(chargedToken);
}
else
{
	if (chargedToken == Actor.InvalidConditionToken)
		chargedToken = self.GrantCondition(Info.ChargedCondition);
}
```

Happy to take either though.

**Review by pchote — DISMISSED** [review]

File: OpenRA.Mods.Cnc/Graphics/ModelRenderable.cs [hunk 2]
```diff
@@ -37,7 +37,8 @@ public class ModelRenderable : IPalettedRenderable, IModifyableRenderable
 			: this(renderer, models, pos, zOffset, camera, scale,
 				lightSource, lightAmbientColor, lightDiffuseColor,
 				color, normals, shadow, 1f,
-				float3.Ones, TintModifiers.None) { }
+				float3.Ones, TintModifiers.None)
+		{ }
```
Comment (penev92) [inline OpenRA.Mods.Cnc/Graphics/ModelRenderable.cs:41]: This one I am not a big fan of. Is there anything we can do?

File: OpenRA.Mods.Cnc/Traits/TDGunboat.cs [hunk 3]
```diff
@@ -196,18 +196,24 @@ public void SetPosition(Actor self, WPos pos)
 		}
 
 		public Activity MoveTo(CPos cell, int nearEnough = 0, Actor ignoreActor = null,
-			bool evaluateNearestMovableCell = false, Color? targetLineColor = null) { return null; }
+			bool evaluateNearestMovableCell = false, Color? targetLineColor = null)
+		{ return null; }
```
Comment (penev92) [inline OpenRA.Mods.Cnc/Traits/TDGunboat.cs:200]: Same here. Looked better before, although they should probably become lambdas already.

File: OpenRA.Mods.Cnc/Graphics/ModelRenderable.cs (hunk 2 above)
Comment (PunkPun) [reply OpenRA.Mods.Cnc/Graphics/ModelRenderable.cs:41]: having constructors be forced to have newline braces doesn't bother me

Comment (RoosterDragon) [reply OpenRA.Mods.Cnc/Graphics/ModelRenderable.cs:41]: The option `csharp_preserve_single_line_blocks = true` is allowing items that are wholly one one line such as `Method(int a) { }` to be preserved.

But as the parameters are already wrapped here, the `csharp_new_line_before_open_brace = all` option is jumping in to say the method body must be wrapped onto a new line.

There aren't any options to treat this differently that I can see. If you're not a fan of the style change, maybe consider if it is "acceptable losses" for enforcing the rule as a whole.

File: OpenRA.Mods.Cnc/Traits/TDGunboat.cs (hunk 3 above)
Comment (RoosterDragon) [reply OpenRA.Mods.Cnc/Traits/TDGunboat.cs:200]: Same issue as above.

Comment (PunkPun) [reply OpenRA.Mods.Cnc/Traits/TDGunboat.cs:200]: these braces should be unwrapped into 3 lines. Thought again this is TDGunboat, a hack trait already

**Review by penev92 — APPROVED** [review]
I am not a big fan of some aspects of this rule, but I'll take it for the greater good, thank you 👍

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/591d9c13ed0377a1a4f9aa714e361cb6440e6857...b5186bcc5939af2620b9c1e6774bc465e8c55ccb)

**Pull request merged**